lms/djangoapps/django_comment_client/management/commands/tests/test_rename_user.py

```diff
@@ -0,0 +1,128 @@
+            "closed": False,
+            "comment_count": 9,
+            "commentable_id": "video_1",
+            "course_id": "MITx/6.002x/2012_Fall",
+            "created_at": datetime.datetime.strptime('22092013', "%d%m%Y"),
+            "historical_abuse_flaggers": [],
+            "last_activity_at": datetime.datetime.strptime('22092013', "%d%m%Y"),
+            "tags_array": [
+                "c#",
+                "2012",
+                "java-sucks"
+            ],
+            "thread_type": "discussion",
+            "title": "Dolore et molestias quia maxime eaque quos voluptatem et.",
+            "updated_at": datetime.datetime.strptime('22092013', "%d%m%Y"),
+            "visible": True,
+            "votes": {
+                "count": 7,
+                "down": [
+                    "1",
+                    "3",
+                    "4",
+                    "5",
+                    "7"
+                ],
+                "down_count": 5,
+                "point": -3,
+                "up": [
+                    "2",
+                    "6"
+                ],
+                "up_count": 2
+            }
+
+        }
+        cls.mongo_comment_ids.append(cls.db.contents.insert(comment_1))
+
+    @classmethod
+    def seed_mongo_user(cls):
+        '''
+        Seeds 3 mongo test users for the tests that follow
+        '''
+        user_1 = {
+            "default_sort_key": "date",
+            "notification_ids": [],
+            "external_id": "test_user_1",
+            "username": cls.sql_user.username
+        }
+        cls.mongo_user_ids.append(cls.db.users.insert(user_1))
+
+    def test_data_seed(self):
+        self.assertTrue(len(self.mongo_user_ids) == 1, 'Test mongo users were not properly seeded, rendering tests invalid.')
+        self.assertTrue(len(self.mongo_comment_ids) == 1, 'Test mongo comments were not properly seeded, rendering the tests invalid.')
+
+    def test_incorrect_num_args(self):
+        zero_args = []
+        one_arg = ['arg']
+        three_args = ['arg', 'arg', 'arg']
+        self.assertRaises(CommandError, self.rename_cmd.handle, *zero_args)
+        self.assertRaises(CommandError, self.rename_cmd.handle, *one_arg)
+        self.assertRaises(CommandError, self.rename_cmd.handle, *three_args)
+
+    def test_successful_change(self):
+        saved_stdout = sys.stdout
+        try:
+            out = StringIO()
+            sys.stdout = out
+            args = [self.sql_user.username, 'someArbitraryUserName']
```

Comment:
For tests like this, where I need two different values (like
`username_old` and `username_new` here), I like to store one value and
then calculate the second from the first. In this case, we would do
something like this:
```python
username_old = self.sql_user.username
username_new = self.sql_user.username + '2'
self.rename_cmd.handle(username_old, username_new)
```

Besides being one less string to keep around (`'someArbitraryUserName'`),
this form actually guarantees that the new and old values will be
unique. The current implementation would fail if
`self.sql_user.username` were set to `'someArbitraryUserName'`.